core/lib/zksync_core/src/api_server/web3/namespaces/zks.rs

```diff
@@ -646,4 +646,14 @@ impl ZksNamespace {
             storage_proof,
         })
     }
+
+    #[tracing::instrument(skip_all)]
+    pub async fn get_base_token_l1_address_impl(&self) -> Result<Address, Web3Error> {
+        const METHOD_NAME: &str = "get_base_token_l1_address_impl";
+        let l1_token_env_var = env::var("CONTRACTS_BASE_TOKEN_ADDR")
+            .map_err(|_err| internal_error(METHOD_NAME, "Missing env variable for base token"))?;
+        Ok(H160::from_str(&l1_token_env_var).map_err(|_err| {
+            internal_error(METHOD_NAME, "Set value for base token is not an address")
+        })?)
```

Comment:
Please do not access environment variables directly from within any module.
Environment variables are not the only way to define the configuration for the application, use a dedicated configuration approach for that.
You may have a look at how other methods that return contract addresses are implemented in this file.

Reply:
Done! Please let me know if there's something else to improve.